sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/DeleteRepositoryResult.java

```diff
@@ -4,12 +4,12 @@
 
 package com.azure.containers.containerregistry.models;
 
-import com.azure.core.annotation.Fluent;
+import com.azure.core.annotation.Immutable;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import java.util.List;
 
 /** Deleted repository. */
-@Fluent
+@Immutable
```

Comment:
Creating immutable model types should have first class support in auto-gen via parameters. As far as I could find the only way to do this today is by writing our own customization layer and passing it to autorest which may become more work based on number of models.

Reply:
This is supported in autorest. If all fields in the model are read-only, then autorest will generate this as immutable type. See [example](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/models/PhoneNumberSearchResult.java#L13). The swagger has to mark these properties as `readonly`.

Reply:
Ahh good to know, will update this in the next iteration - as I already need to consume other swagger changes.

---
In reply to: [609625049](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r609625049) [](ancestors = 609625049)